SeaPkg/Include/Library/PeCoffValidationLib.h

```diff
@@ -0,0 +1,117 @@
+/** @file
+  Defines function definitions for validating a loaded PE/COFF image post execution
+  against an auxiliary file (defined in SeaAuxiliary.h) that contains auxiliary
+  multiple validation entries. A validation entry specifies a specific region in the
+  target image buffer to validate against, a validation type to perform, and any
+  necessary data to assist in the validation.
+
+  There currently exists five validation types:
+  - Non-Zero: Validates that the specified region in the target image buffer is not all zero.
+  - Content: Validates that the specified region in the target image buffer matches the content in the reference data.
+  - MemAttr: Validates that the specified region in the target image buffer belongs to a user page that is mapped inside MM
+             and the page attributes match the requirements specified by the validation entry.
+  - SelfRef: Validates that the specified region in the target image buffer matches the content in the original image buffer as
+             specified by TargetOffset in the validation entry.
+  - Pointer: Validates that the specified region in the target image buffer is a pointer, and that the pointer is not NULL.
+
+  Copyright (c) Microsoft Corporation.
+  SPDX-License-Identifier: BSD-2-Clause-Patent
+
+**/
+
+#ifndef BASE_PECOFF_VALIDATION_LIB_H_
+#define BASE_PECOFF_VALIDATION_LIB_H_
+
+/**
+  Validates a specific region in the target image buffer denoted by [Hdr->Offset: Hdr->Offset + Hdr->Size]
+  is not a zero buffer.
+
+  @param[in] TargetImage  The pointer to the target image buffer.
+  @param[in] Hdr          The header of the validation entry.
+
+  @return EFI_SUCCESS             The target image passes the validation.
+  @return EFI_SECURITY_VIOLATION  The specified buffer in the target image is all zero.
+**/
+EFI_STATUS
+PeCoffImageValidationNonZero (
+  IN VOID                           *TargetImage,

```

Comment:
I know these functions already existed but can some of the params be tightened up in interfaces where possible? Like `CONST VOID *` to better reflect usage.

Reply:
I've updated all pointer usage across all of the validation functions to be `CONST` as they are all read only in their usage. This mainly pertains to the `*TargetImage` and `*Hdr`, but also any other pointers.